linkerd/app/core/src/tls_info.rs

```diff
@@ -0,0 +1,83 @@
+use linkerd_metrics::prom;
+use prometheus_client::encoding::{EncodeLabelSet, EncodeLabelValue, LabelValueEncoder};
+use std::fmt::{Error, Write};
+use std::sync::{Arc, OnceLock};
+use tracing::info;
+
+static TLS_INFO: OnceLock<Arc<TlsInfo>> = OnceLock::new();
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq, EncodeLabelSet)]
+pub struct TlsInfo {
+    tls_suites: MetricValueList,
+    tls_kx_groups: MetricValueList,
+    tls_rand: String,
+    tls_key_provider: String,
+    tls_fips: bool,
+}
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq)]
+struct MetricValueList {
+    values: Vec<&'static str>,
+}
+
+impl FromIterator<&'static str> for MetricValueList {
+    fn from_iter<T: IntoIterator<Item = &'static str>>(iter: T) -> Self {
+        MetricValueList {
+            values: iter.into_iter().collect(),
+        }
+    }
+}
+
+impl EncodeLabelValue for MetricValueList {
+    fn encode(&self, encoder: &mut LabelValueEncoder<'_>) -> Result<(), Error> {
+        for value in &self.values {
+            value.encode(encoder)?;
+            encoder.write_char(',')?;
+        }
+        Ok(())
+    }
+}
+
+pub fn metric() -> prom::Family<TlsInfo, prom::ConstGauge> {
+    let fam = prom::Family::<TlsInfo, prom::ConstGauge>::new_with_constructor(|| {
+        prom::ConstGauge::new(1)
+    });
+    if let Some(tls_info) = TLS_INFO.get() {
+        let _ = fam.get_or_create(tls_info);
+        return fam;
+    }
+
+    let Some(provider) = tokio_rustls::rustls::crypto::CryptoProvider::get_default() else {
+        // If the crypto provider hasn't been initialized, we return the metrics family with an
+        // empty set of metrics.
+        return fam;
+    };
```

Comment:
i'm of two minds on this: returning here seems like a conservative choice, but looking at the call-site gives me some concern.

if the crypto provider hasn't been initialized, is that a bug? we don't seem to call this repeatedly, so if the provider is set up after we call this function, we wouldn't report on it correctly.

Reply:
ah! i'll register my disposition that i have a somewhat strong preference for returning an error here, but i shan't block on it. an error log gets us an indicator of some sort, which is my primary concern :bowing_woman:

Reply:
(soft request to file an issue tracking this follow-on should we proceed to merge as-is!)